ee/control/consumers/tokenconsumer/ingest_token_consumer.go

```diff
@@ -0,0 +1,41 @@
+package tokenconsumer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+const IngestSubsystem = "ingest"
+
+var ObservabilityIngestTokenKey = []byte("observability_ingest_server_token")
+
+type IngestTokenConsumer struct {
+	store types.KVStore
+}
+
+type traceSubsystemConfig struct {
+	IngestToken string `json:"ingest_token"`
+}
+
+func NewIngestTokenConsumer(store types.KVStore) *IngestTokenConsumer {
+	return &IngestTokenConsumer{
+		store: store,
+	}
+}
+
+// Update satisfies control.consumer interface
+func (i *IngestTokenConsumer) Update(data io.Reader) error {
+	var updatedCfg traceSubsystemConfig
```

Comment:
I think we ought to just unmarshal to something like

`map[string]string`

and iterate over the kvps setting the new keys

Reply:
Moved to KeyValue consumer, no need for this anymore!